testnet/stacks-node/src/tests/signer.rs

```diff
@@ -0,0 +1,315 @@
+use std::thread::sleep;
+use std::time::Duration;
+use std::{env, thread};
+use std::sync::mpsc::{channel, Receiver};
+
+use crate::{
+    config::{Config as NeonConfig, EventKeyType, EventObserverConfig, InitialBalance},
+    neon,
+
+    tests::{
+        bitcoin_regtest::BitcoinCoreController,
+        make_contract_publish,
+        neon_integrations::{
+            neon_integration_test_conf, next_block_and_wait, submit_tx,
+            wait_for_runloop,
+        },
+        to_addr,
+    },
+    BitcoinRegtestController, BurnchainController,
+};
+use clarity::vm::types::QualifiedContractIdentifier;
+use libsigner::{RunningSigner, Signer, StackerDBEventReceiver};
+use p256k1::{ecdsa, point::Point, scalar::Scalar};
+use rand_core::OsRng;
+use stacks::chainstate::stacks::StacksPrivateKey;
+use stacks_signer::runloop::RunLoopCommand;
+
+const SLOTS_PER_USER: u32 = 16;
+
+// Helper struct for holding the btc and stx neon nodes
+#[allow(dead_code)]
+struct RunningNodes {
+    pub btc_regtest_controller: BitcoinRegtestController,
+    pub btcd_controller: BitcoinCoreController,
+    pub join_handle: thread::JoinHandle<()>,
+    pub conf: NeonConfig,
+}
+
+fn build_contract(num_signers: u32, signer_stacks_private_keys: &[StacksPrivateKey]) -> String {
```

Comment:
Could `build_contract` and `build_signer_config_tomls` be dropped here, in favor of the ones in the `stacks-signer` crate?

Reply:
Ah, I see that already happened in @jferrant's follow-on PR.